rosidl_generator_cpp/resource/action__struct.hpp.em

```diff
@@ -0,0 +1,55 @@
+// generated from rosidl_generator_cpp/resource/action__struct.hpp.em
+// generated code does not contain a copyright notice
+
+@#######################################################################
+@# EmPy template for generating <action>__struct.hpp files
+@#
+@# Context:
+@#  - spec (rosidl_parser.ActionSpecification)
+@#    Parsed specification of the .action file
+@#  - subfolder (string)
+@#    The subfolder / subnamespace of the message, usually 'action'
+@#  - get_header_filename_from_msg_name (function)
+@#######################################################################
+@
+@{
+header_guard_parts = [
+    spec.pkg_name, subfolder,
+    get_header_filename_from_msg_name(spec.action_name) + '__struct_hpp']
+header_guard_variable = '__'.join([x.upper() for x in header_guard_parts]) + '_'
+}@
+#ifndef @(header_guard_variable)
+#define @(header_guard_variable)
+
+#include <action_msgs/srv/cancel_goal.hpp>
+#include <action_msgs/msg/goal_info.hpp>
+#include <action_msgs/msg/goal_status_array.hpp>
+#include <@(spec.pkg_name)/action/@(get_header_filename_from_msg_name(spec.action_name))__feedback.hpp>
+#include <@(spec.pkg_name)/action/@(get_header_filename_from_msg_name(spec.action_name))__goal.hpp>
+#include <@(spec.pkg_name)/action/@(get_header_filename_from_msg_name(spec.action_name))__result.hpp>
```

Comment:
@apojomovsky I think we should be consistent with [rosidl_generator_c equivalent template](https://github.com/ros2/rosidl/pull/312/files#diff-e3689b5c0e01a46c94551f3b59cb9dd6): either we hard-code `action` or we pass in `@(subfolder)`.

Reply:
agree, done!